File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        
        if(head == null || head.next == null) return null;
        
        ListNode slow=head, slowBehind=head, fast=head;
        while(fast!=null && fast.next!=null){
            slowBehind = slow;
            slow = slow.next;
            fast = fast.next.next;
        }
        
        slowBehind.next = slow.next;
        return head;
    }
}

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        
        if(head == NULL || head->next == NULL) return NULL;
        
        ListNode* slow = head, *slowBehind = head, *fast = head;
        while(fast != NULL && fast->next != NULL){
            slowBehind = slow;
            slow = slow->next;
            fast = fast->next->next;
        }
        
        slowBehind->next = slow->next;
        delete slow;
        return head;
    }
};